From ec07502b1c1f3f7badf161d006246950670c8580 Mon Sep 17 00:00:00 2001
From: Marc Jones <marcj303@gmail.com>
Date: Wed, 9 Jun 2021 12:44:57 -0600
Subject: [PATCH 3/3] mainboard/ocp/monolake: Expand BIOS region

Expand the BIOS region by 1MB for larger LinuxBIOS payloads. This
shrinks the ME region by 1MB.

Warning: This also requires a matching change to the firmware descriptor
and ME binary.

Change-Id: I4161c920de2f7cbafa103bb2068bda50ac02e992
Signed-off-by: Marc Jones <marcj303@gmail.com>
---
 src/mainboard/ocp/monolake/board.fmd    |  8 ++++----
 src/mainboard/ocp/monolake/vboot-ro.fmd | 10 +++++-----
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/src/mainboard/ocp/monolake/board.fmd b/src/mainboard/ocp/monolake/board.fmd
index d0265e464d..af9b7f7e94 100644
--- a/src/mainboard/ocp/monolake/board.fmd
+++ b/src/mainboard/ocp/monolake/board.fmd
@@ -1,9 +1,9 @@
 FLASH@0xff000000 0x1000000 {
-	SI_ALL@0x0 0x800000 {
+	SI_ALL@0x0 0x700000 {
 		SI_DESC@0x0 0x1000
-		SI_ME@0x1000 0x7ff000
+		SI_ME@0x1000 0x6ff000
 	}
-	SI_BIOS@0x800000 0x800000 {
+	SI_BIOS@0x700000 0x900000 {
 		FMAP@0x0 0x1000
 		RW_MISC@0x1000 0x9000 {
 			RW_ELOG@0x0 0x4000
@@ -17,6 +17,6 @@ FLASH@0xff000000 0x1000000 {
 		}
 		RW_MRC_CACHE@0xE000 0x10000
 		CONSOLE@0x1E000 0x10000
-		COREBOOT(CBFS)@0x2E000 0x7d2000
+		COREBOOT(CBFS)@0x2E000 0x8d2000
 	}
 }
diff --git a/src/mainboard/ocp/monolake/vboot-ro.fmd b/src/mainboard/ocp/monolake/vboot-ro.fmd
index 44be3370ce..eaccef00f8 100644
--- a/src/mainboard/ocp/monolake/vboot-ro.fmd
+++ b/src/mainboard/ocp/monolake/vboot-ro.fmd
@@ -1,21 +1,21 @@
 FLASH 16M {
-	SI_ALL@0x0 0x800000 {
+	SI_ALL@0x0 0x700000 {
 		SI_DESC@0x0 0x1000
-		SI_ME@0x1000 0x7ff000
+		SI_ME@0x1000 0x6ff000
 	}
-	SI_BIOS@0x800000 0x800000 {
+	SI_BIOS@0x700000 0x900000 {
 		MISC_RW@0x0 0x20000 {
 			RW_MRC_CACHE@0x0 0x10000
 			RW_VPD(PRESERVE)@0x010000 0x4000
 		}
-		WP_RO@0x020000 0x7e0000 {
+		WP_RO@0x020000 0x8e0000 {
 			RO_VPD(PRESERVE)@0x0 0x4000
 			RO_SECTION@0x4000 0x7dc000 {
 				FMAP@0x0 0x800
 				RO_FRID@0x800 0x40
 				RO_FRID_PAD@0x840 0x7c0
 				GBB@0x1000 0x4000
-				COREBOOT(CBFS)@0x5000 0x7d7000
+				COREBOOT(CBFS)@0x5000 0x8d7000
 			}
 		}
 	}
-- 
2.25.1

